htmresearch/frameworks/nlp/classification_model.py

```diff
@@ -19,335 +19,183 @@
-            ('B', ('import this', array([0, 3]), '1')
-        ])
+  def inferToken(self, token, reset=0, returnDetailedResults=False,
+                 sortResults=True):
     """
-    outDict = OrderedDict()
-    for dataID, data in dataDict.iteritems():
-      outDict[dataID] = (self.prepText(data[0], preprocess), data[1], data[2])
+    Classify the token (i.e. run inference on the model with this document) and
+    return classification results and (optionally) a list of sampleIds and
+    distances.   Repeated sampleIds are NOT removed from the results.
 
-    return outDict
+    @param token    (str)     The text token to train on
+    @param reset    (int)     Should be 0 or 1. If 1, assumes we are at the
+                              end of a sequence. A reset signal will be issued
+                              after the model has been trained on this token.
+    @param returnDetailedResults
+                    (bool)    If True will return sampleIds and distances
+                              This could slow things down depending on the
+                              number of stored patterns.
+    @param sortResults (bool) If True the list of sampleIds and distances
+                              will be sorted in order of increasing distances.
 
-
-  @staticmethod
-  def prepText(text, preprocess=False):
-    """
-    Returns a list of the text tokens.
-
-    @param preprocess   (bool)    Whether or not to preprocess the text data.
+    @return  (numpy array) An array of size numLabels. Position i contains
+                           the likelihood that this token belongs to the
+                           i'th category. An array containing all zeros
+                           implies no decision could be made.
+             (list)        A list of sampleIds
+             (numpy array) An array of distances from each stored sample
     """
-    if preprocess:
-      sample = TextPreprocess().tokenize(text,
-                                         ignoreCommon=100,
-                                         removeStrings=["[identifier deleted]"],
-                                         correctSpell=True)
-    else:
-      sample = TextPreprocess().tokenize(text)
+    # TODO: Should we specify that distances normalized between 0 and 1?
+    # Currently we use whatever the KNN returns.
 
-    return sample
+    raise NotImplementedError
 
 
-  def tokenize(self, text, preprocess=False):
+  def inferDocument(self, document, returnDetailedResults=False,
+                    sortResults=True):
     """
-    Given a bunch of text (could be several sentences) return a single list
-    containing individual tokens.
+    Run inference on the model with this document and return classification
+    results, sampleIds and distances.  By default this routine will tokenize the
+    document and classify using these tokens. A reset is issued after inference.
+    Repeated sampleIds ARE removed from the results.
```

Comment:
Please note the classification method here is to get the single most likely category for each token and add 1 to the `categoryVotes` array.